packages/msgraph/src/Utils.js

```diff
@@ -1,13 +1,31 @@
+import { fetch } from 'undici';
 import { composeNextState } from '@openfn/language-common';
 
-export function setUrl(urlParams) {
-  const { apiVersion, resolvePath } = urlParams;
+export function ensureDriveExistsOrFail(drive, fileType) {
+  if (!drive) {
+    const operation = fileType === 'folder' ? 'getFolder' : 'getFile';
+    const egJobCode = [
+      `Eg: Get a site drive then get "/Sample Data" ${fileType}`,
```

Comment:
I really like this level of help but I do think it's a bit over-complicated. That the `fileType` parameter is kinda weird and won't scale. I'd like to remove it to simplify the signature.

I think I'd make the whole error message just say this:
```
Drive is not defined.

At the top of your job you should define all the drives you want to us.

eg: getDrive({ id: "openfn.sharepoint.com", owner: "sites"})
```
if we change the parameters  to `state, driveName`, you should take the drive name into account (which is probably a different example)/

Reply:
Done here [839975e](https://github.com/OpenFn/adaptors/pull/297/commits/839975e9adb3311d10912eb66c1a419ae55960db)